fontbe/src/features.rs

```diff
@@ -111,6 +113,190 @@ impl<'a> FeaVariationInfo<'a> {
+                    .iter()
+                    .map(|name| GlyphId::new(self.glyph_map.glyph_id(name).unwrap_or(0) as u16))
+                    .collect();
+                (class_name, glyph_class)
+            })
+            .collect::<HashMap<_, _>>();
+
+        let mut ppos_subtables = PairPosBuilder::default();
+
+        // now for each kerning entry, directly add a rule to the builder:
+        for ((left, right), values) in &self.kerning.kerns {
+            let (default_value, deltas) = self
+                .resolve_variable_metric(values)
+                .expect("FIGURE OUT ERRORS");
+
+            let mut x_adv_record = ValueRecord::new().with_x_advance(default_value);
+            let empty_record = ValueRecord::default();
+
+            if !deltas.is_empty() {
+                let var_idx = builder.add_deltas(deltas);
+                x_adv_record = x_adv_record.with_x_advance_device(var_idx);
+            }
+
+            match (left, right) {
+                (KernParticipant::Glyph(left), KernParticipant::Glyph(right)) => {
+                    let gid0 = name_to_gid(left);
+                    let gid1 = name_to_gid(right);
+                    ppos_subtables.insert_pair(gid0, x_adv_record, gid1, empty_record);
+                }
+                (KernParticipant::Group(left), KernParticipant::Group(right)) => {
+                    let left = glyph_classes.get(left).unwrap().clone();
+                    let right = glyph_classes.get(right).unwrap().clone();
+                    ppos_subtables.insert_classes(left, x_adv_record, right, empty_record);
+                }
+                // if groups are mixed with glyphs then we enumerate the group
+                (KernParticipant::Glyph(left), KernParticipant::Group(right)) => {
+                    let gid0 = name_to_gid(left);
+                    let right = glyph_classes.get(right).unwrap();
+                    for gid1 in right {
+                        ppos_subtables.insert_pair(
+                            gid0,
+                            x_adv_record.clone(),
+                            *gid1,
+                            empty_record.clone(),
+                        );
+                    }
+                }
+                (KernParticipant::Group(left), KernParticipant::Glyph(right)) => {
+                    let left = glyph_classes.get(left).unwrap();
+                    let gid1 = name_to_gid(right);
+                    for gid0 in left {
+                        ppos_subtables.insert_pair(
+                            *gid0,
+                            x_adv_record.clone(),
+                            gid1,
+                            empty_record.clone(),
+                        );
+                    }
+                }
+            }
+        }
+
+        // now we have a builder for the pairpos subtables, so we can make
+        // a lookup:
+        let lookup_id = builder.add_lookup(LookupFlag::empty(), None, vec![ppos_subtables]);
```

Comment:
ufo2ft kernFeatureWriter sets IgnoreMarks lookupflag by default on its kerning lookups, unless there are kerning pairs between glyphs classified respectively as bases and marks (or marks vs marks); the latter are split into separate lookups where IgnoreMarks lookupflag is _not_ set.